### PR TITLE
bittrex: handle errors

### PR DESCRIPTION
**Pull request by mkutny**
Bittrex returns HTTP 200 on errors but `handleErrors` expected ` >= 400`.

As such, error handling didn't work for unknown errors:
```
ccxt> node examples/js/cli.js bittrex createLimitBuyOrder ETH/BTC 1 
bittrex.createLimitBuyOrder (ETH/BTC, 1, 0)
---------------------------------------------------
[TypeError] Cannot read property 'result' of undefined

    at createOrder    js/bittrex.js:447                  'id': response['result'][orderIdField],
    at _tickCallback  internal/process/next_tick.js:160                                         
```

### CODE REVIEW

**Comment by kroitor** [comment]
Thx! ) Good to hear from you)

File: js/bittrex.js [hunk 1]
```diff
@@ -136,6 +136,19 @@ module.exports = class bittrex extends Exchange {
...
+                'INVALID_SIGNATURE': AuthenticationError,
+                'INVALID_CURRENCY': ExchangeError,
+                'INVALID_PERMISSION': AuthenticationError,
+                'INSUFFICIENT_FUNDS': InsufficientFunds,
+                'QUANTITY_NOT_PROVIDED': InvalidOrder,
+                'MIN_TRADE_REQUIREMENT_NOT_MET': InvalidOrder,
+                'ORDER_NOT_OPEN': InvalidOrder,
+                'UUID_INVALID': OrderNotFound,
+                'RATE_NOT_PROVIDED': InvalidOrder, // createLimitBuyOrder ('ETH/BTC', 1, 0)
+                'Insufficient funds!': InsufficientFunds, // bluetrade
```
Comment (kroitor) [inline js/bittrex.js]: I'd move this line to bluetrade.js upon merging, if you don't mind )

Comment (mkutny) [reply js/bittrex.js]: Oops... completely overlooked it.
Actually, bleutrade error messages seem to be quite different anyway, added some of them.

Comment (kroitor) [reply js/bittrex.js]: Thx for fixing it, I didn't mean to force it on you, was going to merge %) anyways, thanks for the added messages as well.

Comment (mkutny) [reply js/bittrex.js]: Your questions just forced me to think that I completely forgot to test bleutrade, so I had to revert to it anyway.

File: js/bittrex.js [hunk 2]
```diff
@@ -690,30 +690,28 @@ module.exports = class bittrex extends Exchange {
...
+            if (typeof success === 'undefined')
+                throw new ExchangeError (this.id + ': malformed response: ' + this.json (response));
+            if (typeof success === 'string')
+                // bleutrade uses string instead of boolean
+                success = (success === 'true') ? true : false;
+            if (!success) {
+                const message = this.safeString (response, 'message');
+                const feedback = this.id + ' ' + this.json (response);
+                const exceptions = this.exceptions;
+                if (message in exceptions)
```
Comment (mkutny) [inline js/bittrex.js]: I don't check if `message` is `undefined` because JS allows looking up `undefined` properties. Will it work in Python/PHP too?

Comment (kroitor) [reply js/bittrex.js]:
```
mbp:ccxt igorkroitor$ python
Python 2.7.13 (default, Jun  5 2017, 10:04:07)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 'a' if None in {'foo':'bar'} else 'b'
'b'
```

Comment (kroitor) [reply js/bittrex.js]:
```PHP
php > var_dump (array_key_exists(undefined, array('foo' => 'bar')));
bool(false)
```

Comment (mkutny) [reply js/bittrex.js]: Thanks! )

**Comment by rdbr** [comment]
Thanks for the solution! This is what I meant in #2137

**Pull request merged**